snap/naming/componentref_test.go

```diff
@@ -85,3 +88,103 @@ func (s *componentRefSuite) TestSplitFullComponentNameErr(c *C) {
 		c.Check(comp, Equals, "")
 	}
 }
+
+func (s *componentRefSuite) TestComponentRefFromSnapPackFilename(c *C) {
+	type test struct {
+		filename string
+		cr       naming.ComponentRef
+		fail     bool
```

Comment:
nitpick, maybe you could just check if cr is empty instead of adding a fail flag. Anyway, if it is an error case this should check the returned ref is empty.